app/assets/javascripts/dashboard/views/dataset_view.js

```diff
@@ -30,14 +31,31 @@ type State = {
   isLoading: boolean,
 };
 
-function createThumbnailURL(datasetName: string, layers: Array<APIDatasetType>): string {
+function createThumbnailURL(datasetName: string, layers: Array<DataLayerType>): string {
   const colorLayer = _.find(layers, { category: "color" });
   if (colorLayer) {
     return `/api/datasets/${datasetName}/layers/${colorLayer.name}/thumbnail`;
   }
   return "";
 }
 
+export function transformDatasets(datasets: Array<DatasetType>): Array<DatasetType> {
```

Comment:
Shouldn't the function take an array of `APIDatasetType`? Since it adds exactly the three properties which are new in `DatasetType` in comparison to `APIDatasetType`...